subprojects/build-events/src/main/java/org/gradle/internal/build/event/types/DefaultFileComparisonTestAssertionFailure.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.internal.build.event.types;
+
+import org.gradle.tooling.internal.protocol.InternalFailure;
+import org.gradle.tooling.internal.protocol.InternalFileComparisonTestAssertionFailure;
+
+import java.util.Collections;
+import java.util.List;
+
+public class DefaultFileComparisonTestAssertionFailure extends DefaultTestAssertionFailure implements InternalFileComparisonTestAssertionFailure {
+
+    private final byte[] expectedContent;
+    private final byte[] actualContent;
```

Comment:
🤔 Is storing the entire file in memory a good choice?
I'm sure somebody in the universe will have a GB/TB-sized file for some reason, and you will need to keep it in the memory.